src/store/attachment.js

```diff
@@ -0,0 +1,80 @@
+/*
+ * @copyright Copyright (c) 2018 Julius Härtl <jus@bitgrid.net>
+ *
+ * @author Julius Härtl <jus@bitgrid.net>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+import { AttachmentApi } from './../services/AttachmentApi'
+import Vue from 'vue'
+
+const apiClient = new AttachmentApi()
+
+export default {
+	state: {
+		attachments: {},
+	},
+	getters: {
+		attachmentsByCard: state => (cardId) => {
+			if (state.attachments[cardId] === undefined) {
+				return []
+			}
+			return state.attachments[cardId]
+		},
+
+	},
+	mutations: {
+		createAttachment(state, { cardId, attachments }) {
+			if (state.attachments[cardId] === undefined) {
+				Vue.set(state.attachments, cardId, attachments)
+			} else {
+				state.attachments[cardId].push(attachments)
+			}
+			// let card = getters.cardById(cardId)
+			// console.log(card)
+
+		},
+
+		deleteAttachment(state, deletedAttachment) {
+			const existingIndex = state.attachments[deletedAttachment.cardId].findIndex(a => a.id === deletedAttachment.id)
+			if (existingIndex !== -1) {
+				state.attachments[deletedAttachment.cardId].splice(existingIndex, 1)
+			}
+		},
+
+	},
+	actions: {
+		async getAllAttachments({ commit }, cardId) {
```

Comment:
I would prefer a naming that is more like a action, maybe fetchAttachments, as I think get might lead to confusion that it could be a getter